okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -199,6 +220,31 @@
     this.socket = Preconditions.checkNotNull(socket);
     this.outboundFlow = new OutboundFlowController(this, frameWriter);
     this.nextStreamId = nextStreamId;
+    this.ticker = ticker;
+  }
+
+  @Override
+  public void ping(final PingCallback callback, Executor executor) {
+    checkState(frameWriter != null);
+    long data;
+    Http2Ping p;
+    synchronized (lock) {
+      if (goAwayStatus != null) {
```

Comment:
Maybe we should use `stopped` here instead? GoAwayStatus is not guaranteed to be non-`null` though, so we would have to generate a status for that case.